src/core/thread/mle_ftd.cpp

```diff
@@ -3529,6 +3529,11 @@ void Mle::ProcessAddressSolicit(AddrSolicitInfo &aInfo)
     aInfo.mResponse = kAddrSolicitNoAddressAvailable;
     aInfo.mRouter   = nullptr;
 
+    // The leader may have chosen to begin the attachment process assuming it's own partition
+    // was a singleton. Don't allow any devices to upgrade while it is in the attaching
+    // process because it could change the preconditions for the decision to reattach.
+    VerifyOrExit(!IsAttaching());
```

Comment:
It seems that this behavior is not specifically called out or prevented by spec at this time. One question I had was whether or not it made sense to add another response code for this. Other devices treat any response that isn't "success" as failure.

Reply:
This is a valid point.

As a suggestion, let's add this check in the `void Mle::HandleTmf<kUriAddressSolicit>()` method instead. We can change `VerifyOrExit(IsLeader());` to `VerifyOrExit(IsLeader() && !IsAttaching());`.

This way, we skip processing the request and will not send an Address Solicit Response back to the requesting device. This is better than sending an error status, which it would treat as a "rejection" by the leader and might not try again.

Reply:
Hmm, this made me look into what happens on a REED that never receives a response to an address solicit message.

It seems like mAddressSolicitPending is only cleared when receiving an address solicit response (or upon a reset), so I think this would mean that devices would get stuck today unable to upgrade if no response is sent because the transmit address solicit function verifies that one is not already pending.